khelo/lib/ui/flow/score_board/components/over_complete_sheet.dart

```diff
@@ -0,0 +1,225 @@
+    bool showBg = true,
+    bool isHeading = false,
+  }) {
+    return Container(
+      width: double.infinity,
+      padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 14),
+      decoration: BoxDecoration(
+          color: showBg
+              ? isHeading
+                  ? context.colorScheme.containerNormal
+                  : context.colorScheme.surface
+              : Colors.transparent,
+          borderRadius: BorderRadius.circular(8)),
+      child: Text.rich(
+          maxLines: 2,
+          overflow: TextOverflow.ellipsis,
+          TextSpan(
+              text: placeholder,
+              style: AppTextStyle.body2
+                  .copyWith(color: context.colorScheme.textDisabled),
+              children: [
+                WidgetSpan(
+                    child: SizedBox(
+                  width: placeholder.isEmpty ? 0 : 16,
+                )),
+                TextSpan(
+                    text: title,
+                    style: isHeading
+                        ? AppTextStyle.subtitle2
+                            .copyWith(color: context.colorScheme.textPrimary)
+                        : AppTextStyle.body2
+                            .copyWith(color: context.colorScheme.textPrimary)),
+              ])),
+    );
+  }
+
+  Widget _batsManIndividualScore(
+    BuildContext context,
+    ScoreBoardViewState state,
+    MatchPlayer batsMan,
+  ) {
+    final (run, ball) = _getBatsManTotalRuns(state, batsMan.player.id);
+    return _detailInfoCell(
+      context,
+      placeholder: batsMan.player.name ?? "",
+      title: "$run($ball)",
+      showBg: false,
+    );
+  }
+
+  (int, int) _getBatsManTotalRuns(ScoreBoardViewState state, String batsManId) {
+    final scoresList = state.currentScoresList
+        .where((element) => element.batsman_id == batsManId);
+
+    int totalRuns = scoresList
+        .where((element) => (element.extras_type == ExtrasType.noBall ||
+            element.extras_type == null))
+        .fold(0, (sum, element) => sum + element.runs_scored);
+
+    final batsManFacedBall = scoresList
+        .where((element) => (element.extras_type != ExtrasType.wide))
+        .length;
+
+    return (totalRuns, batsManFacedBall);
+  }
+
+  (int, int, int) _getCurrentOverStatics(ScoreBoardViewState state) {
+    final scoresList = state.currentScoresList
+        .where((element) => element.over_number == state.overCount);
+
+    int run = scoresList
+        .where((element) => (element.extras_type == ExtrasType.noBall ||
+            element.extras_type == null))
+        .fold(0, (sum, element) => sum + element.runs_scored);
+
+    final wicket = scoresList
+        .where((element) => (element.wicket_type != null &&
+            element.wicket_type != WicketType.retiredHurt))
+        .length;
+
+    final extras = scoresList
+        .where((element) => (element.extras_type != null))
+        .fold(0, (sum, element) => sum + (element.extras_awarded ?? 0));
```

Comment:
I think we need to create a model class to store the sorting value, contain all sorted values in the ViewModel, and then pass the model here to assign the value.